### ISSUE TITLE
[2.4] Auth configuration of OpenLDAP fails with error "Error creating ssl connection: LDAP Result Code 200 "": EOF" 

### ISSUE THREAD

**Issue by maggieliu** [rancher/rancher#29810]
back port for https://github.com/rancher/rancher/issues/29153

gz#12435

**Comment by mrajashree** [comment]
Backend PRs merged, needs UI: https://github.com/rancher/rancher/issues/30617

**Comment by izaac** [comment]
OpenLDAP was validated in  #30617 

Working on getting AD and FreeIPA with StartTLS.

**Comment by izaac** [comment]
Rancher version: `v2.4-head (01/12/2020)` _8e9977805_
Basic P1 use cases for authentication covered in all cases.

Encryption options:
- None
- TLS
- STARTTLS
![Screen Shot 2021-01-13 at 9 37 01 AM](https://user-images.githubusercontent.com/27659/104481347-ecb00480-5582-11eb-8809-83ec367111d3.png)


I tested the three encryption options with Active Directory and self-signed certs (this option must provide a CA cert for TLS and STARTTLS).
For FreeIPA and OpenLDAP I used test environments that have valid Letsencrypt certs. None, TLS and StartTLS worked.
